VirtoCommerce.Platform.Core/Common/EnumUtility.cs

```diff
@@ -11,12 +11,41 @@ public static class EnumUtility
 		public static T SafeParse<T>(string value, T defaultValue)
 			where T : struct
 		{
-			T result;
+            if (!Enum.TryParse(value, out T result))
+            {
+                result = defaultValue;
+            }
 
-			if (!Enum.TryParse(value, out result))
-				result = defaultValue;
-
-			return result;
+            return result;
 		}
-	}
+
+        public static T SafeParseFlags<T>(string value, T defaultValue)
+            where T : struct
+        {            
+            return SafeParseFlags<T>(value, defaultValue, ",");
+        }
+
+        public static T SafeParseFlags<T>(string value, T defaultValue, string separator)
+            where T : struct
+        {
+            var result = 0;
```

Comment:
Ideally we need to include Enum type constraint in where, but it is available only in C# 7.3, so we could update it then and simpler method without all that Convert.ToInt32 casts. About arbitrary [Flags] attribute - do we really want to limit that functionality? It could successfully parse ordinary enums without being [Flags] now, maybe it could be useful. If we need - i could add the check.

Will remove default value redundant casts.

Reply:
Done